File: scraper/entities/variantImage/VariantImage.py
from lib.database.createDatabaseConnection import db
from pony.orm import PrimaryKey, Required, Optional


class VariantImage(db.Entity):
    _table_ = 'variant_images'
    id = PrimaryKey(int, auto=True)
    src = Required(str)
    width = Required(int)
    height = Required(int)
    variant = Optional('Variant')

File: server/src/entities/variantImage/VariantImage.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToOne } from 'typeorm';
import { VariantType } from '../variant/types';
import { VariantImageType } from './types';

@Entity({ name: 'variant_images' })
export default class VariantImage implements VariantImageType {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column('varchar', { length: 255 })
    src!: string;

    @Column('decimal')
    width!: number;

    @Column('decimal')
    height!: number;

    @OneToOne('Variant', 'id')
    variant!: VariantType;
}
